Tutorials/WinForms/Classic/bind-the-data-grid-to-large-data-source.md

```diff
@@ -0,0 +1,294 @@
+# Bind the Data Grid to a large data source
+
+[Step 3](/implement-crud-functionality-with-xpo-objects.md)
+&nbsp;&nbsp;&nbsp;
+[Back to TOC](../../)
+
+## Create the edit order form
+* Select the **EditCustomerForm** item in the Solution Explorer window and press the **Ctrl+C** keyboard shortcut to copy the Form
+* Press the **Ctrl+V** keyboard shortcut to create an EditCustomerForm copy. Rename it to **EditOrderForm**
+* Change the class name and constructor name accordingly. Open the CustomersListform.Designer.cs (vb) file and rename the class there. If this file is hidden, use the **Show All Files** toolbar item in the Solution Explorer window
+* Open the EditOrderForm designer and change the **Text** property to **Edit Order**
+* Rename the **CustomerBindingSource** component to **OrderBindingSource**
+* Rebuild the project and set the **OrderBindingSource.ObjectClassInfo** property to **DxSample.DataAccess.Order** (select the corresponding item in the combo box)
+* Set the **OrderBindingSource.DisplayableProperties** property to **ProductName;OrderDate;Freight;Customer!Key**. The last property name (**Customer!Key** is a [virtual property](https://docs.devexpress.com/XPO/3113/concepts/property-descriptors) used to edit an object reference; see also: [How to: Bind an XPCollection to a LookUp](https://docs.devexpress.com/XPO/2000/examples/how-to-bind-an-xpcollection-to-a-lookup))
+* Rename the **CustomerLayoutControl** component to **OrderLayoutControl**
+* Rebuild the project, select the **OrderLayoutControl** component on the design surface, and click a smart-tag icon at the top-right corner
+* Click the **Retrieve Fields** command in the smart tag menu to open the Select Binding Source wizard.
+* Click the **Next** button and choose the following properties and editors on the Manage Data Bindings screen:
+    * ProductName, TextEdit
+    * OrderDate, DateEdit
+    * Freight, CalcEdit
+    * Customer, LookupEdit 
+* Click the **Finish** button, delete the **First Name** and **Last Name** layout items, reorder the remaining layout items to have the layout similar to the EditCustomerForm, and save changes
+* Add the **XPBindingSource** component to the Form and rename it to **CustomersBindingSource**.
+* Rebuild the project and set properties:
+    * ObjectClassInfo=DxSample.DataAccess.Customer
+    * DisplayableProperties=Oid;ContactName
+* Select the **lookUpEdit1** editor placed near the **Customer** label and open its smart-tag at the top-right corner. Open the combo box near the **Data Source** field, find the **CustomersBindingSource** component under the **Other Data Sources > EditOrderForm List Instances** node, and select it. 
+* Select the **ContactName** and **Oid** properties as the **Display Member** and **Value Member** if they were not selected automatically.
```

Comment:
**Oid**?

Reply:
Yes, this is the real property name ([XPObject.Oid Property](https://docs.devexpress.com/XPO/DevExpress.Xpo.XPObject.Oid)